src/main/resources/js/asmLib.js

```diff
@@ -0,0 +1,200 @@
+                                return result;
+                            }
+
+                            // The following methods only return something other
+                            // than the builder on a particular overload (an overload
+                            // that takes no arguments)
+                            if (argArray.length === 0 && ['astore', 'fstore', 'istore', 'dstore', 'lstore'].includes(key)) {
+                                return result;
+                            }
+
+                            // The method we've called just returns the builder,
+                            // so instead return our modified proxy
+                            return proxy;
+                        }
+                    })
+                }
+            }
+        });
+
+        return proxy;
+    }
+
+    instructions(insnList) {
+        // Wrap the insnList in a proxy so we can "add" methods to the
+        // InsnListBuilder provided by ASMHelper. This proxy delegates
+        // all gets and calls to the target handler except those specified
+        // in the 'get' trap
+        this.insnList = $ => insnList(this.proxyInsnList($));
+
+        return this;
+    }
+
+    execute() {
+        if (this.insnList === undefined) {
+            throw new Error('InjectBuilder requires a call to instructions()');
+        }
+
+        asmInjection(this.className, this.at, this.methodName, this.descriptor, this.fieldMap, this.methodMap, this.insnList);
+    }
+}
+
+export default class ASM {
+    static INTEGER = 'java/lang/Integer';
+    static DOUBLE = 'java/lang/Double';
+    static LONG = 'java/lang/Long';
+    static BOOLEAN = 'java/lang/Boolean';
+    static SHORT = 'java/lang/Short';
+    static CHARACTER = 'java/lang/Character';
+    static BYTE = 'java/lang/Byte';
+    static OBJECT = 'java/lang/Object';
+    static STRING = 'java/lang/String';
+
+    static MINECRAFT = 'net/minecraft/client/Minecraft';
+    static ENTITY = 'net/minecraft/entity/Entity';
+    static ENTITY_FX = 'net/minecraft/client/particle/EntityFX';
+    static ENTITY_ITEM = 'net/minecraft/entity/item/EntityItem';
+    static ENTITY_PLAYER = 'net/minecraft/entity/player/EntityPlayer';
+    static FILE = 'java/io/File';
+    static FRAME_BUFFER = 'net/minecraft/client/shader/Framebuffer';
+    static ICHATCOMPONENT = 'net/minecraft/util/IChatComponent';
+    static INVENTORY_PLAYER = 'net/minecraft/entity/player/InventoryPlayer';
+    static ITEM_STACK = 'net/minecraft/item/ItemStack';
+    static PACKET = 'net/minecraft/network/Packet';
+    static BLOCK_POS = 'net/minecraft/util/BlockPos';
+    static ENUM_FACING = 'net/minecraft/util/EnumFacing';
+    static CANCELLABLE_EVENT = 'com/chattriggers/ctjs/minecraft/listeners/CancellableEvent';
+    static TRIGGER_TYPE = 'com/chattriggers/ctjs/triggers/TriggerType';
+
+    static currentModule = "";
+
+    static JumpCondition = Java.type('me.falsehonesty.asmhelper.dsl.instructions.JumpCondition');

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['Java' is not defined.](https://app.codacy.com/manual/FalseHonesty/ct.js/pullRequest?prid=5138651)